### PR TITLE
add option to collect the count of active memory maps for linux processes

### PR DESCRIPTION
**Pull request by dothebart**
In linux the maximum count of memory mapped files is configured via `/proc/sys/vm/max_map_count` which is 65530 by default. 

With bigger machines that are capable of running many application threads this may quickly become a sparse resource - therefore I'd like to be able to track it.

### CODE REVIEW

File: src/processes.c [hunk 1]
```diff
@@ -738,7 +752,7 @@ static void ps_submit_proc_list(procstat_t *ps) {
   plugin_dispatch_values(&vl);
 
   if ((ps->io_rchar != -1) && (ps->io_wchar != -1)) {
-    sstrncpy(vl.type, "io_octets", sizeof(vl.type));
+    sstrncpy(vl.type, "ps_disk_octets", sizeof(vl.type));
```
Comment (rpv-tomsk) [inline src/processes.c]: Hmm.. Looks as an unrelated change.

File: src/processes.c [hunk 2]
```diff
@@ -761,13 +775,20 @@ static void ps_submit_proc_list(procstat_t *ps) {
     plugin_dispatch_values(&vl);
   }
 
-  if (ps->num_fd > 0) {
+ if (ps->num_fd > 0) {
```
Comment (rpv-tomsk) [inline src/processes.c]: This change also unneeded.

File: src/collectd.conf.in [hunk 3]
```diff
@@ -1170,6 +1170,7 @@
 #<Plugin processes>
 #	CollectFileDescriptor true
 #	CollectContextSwitch true
+#       CollectMemoryMaps true
```
Comment (rpv-tomsk) [inline src/collectd.conf.in]: Please, fix indent.

**Comment by rpv-tomsk** [comment]
Hi! Thanks for your work.

I want to notice, you missed adding documentation for a new option.

File: src/processes.c (hunk 2 above)
Comment (octo) [reply src/processes.c]: I'd like to note that *clang-format* has the final say in this. I'd say it would undo this change.

File: src/types.db [hunk 4]
```diff
@@ -129,6 +129,7 @@ latency                 value:GAUGE:0:U
 links                   value:GAUGE:0:U
 load                    shortterm:GAUGE:0:5000, midterm:GAUGE:0:5000, longterm:GAUGE:0:5000
 memory_bandwidth        value:DERIVE:0:U
+memory_maps             value:GAUGE:0:U
```
Comment (octo) [inline src/types.db]: Would the existing `file_handles` type also work? E.g. as "memory/file_handles-mapped"?

**Comment by octo** [comment]
Thanks @dothebart for submitting this!
Thanks @rpv-tomsk for your review! Is much appreciated!

File: src/types.db (hunk 4 above)
Comment (dothebart) [reply src/types.db]: Ah, I've added this to make it appear on top level in a descriptive name. How would I reference the file_handles type to achieve this?

File: src/processes.c [hunk 5]
```diff
@@ -768,6 +782,13 @@ static void ps_submit_proc_list(procstat_t *ps) {
     plugin_dispatch_values(&vl);
   }
 
+  if (ps->num_maps > 0) {
+    sstrncpy(vl.type, "memory_maps", sizeof(vl.type));
```
Comment (octo) [inline src/processes.c]: > How would I reference the file_handles type to achieve this?

To use the existing `file_handles` type:

```c
sstrncpy(vl.type, "file_handles", sizeof(vl.type));
sstrncpy(vl.type_instance, "mapped", sizeof(vl.type_instance));

**Comment by dothebart** [comment]
any hints on a timeframe when this will occur in a release? weeks? months?

**Comment by octo** [comment]
Weeks. We plan to release 5.8 this year, likely first half of December.
